pkg/discovery/discovery_manager.go

```diff
@@ -97,7 +97,7 @@ func NewManager(logger log.Logger, reg prometheus.Registerer, options ...func(*M
 		metrics:        newMetrics(reg),
 		updatert:       5 * time.Second,
 		triggerSend:    make(chan struct{}, 1),
-		pidLabels:      make(map[int]model.LabelSet),
+		pidLabels:      make(map[uint32]model.LabelSet),
```

Comment:
Process IDs are an `int` (https://ftp.gnu.org/old-gnu/Manuals/glibc-2.2.3/html_node/libc_554.html). I was planning to change all the `uint32`s to become `int`s.

May I suggest that we undo this change and in this, or another PR, change all the PIDs to be 32 bit signed integers?

Reply:
Of course, I thought they are `uint32` :see_no_evil: I'll do the change everywhere. Amazing catch!

Reply:
I'm ok to represent every PID in user-space a `int`. However the PID we get from eBPF helper functions are `uint32` https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#4-bpf_get_current_pid_tgid